omnibus/config/templates/datadog-agent/systemd.service.erb

```diff
@@ -8,6 +8,7 @@ Type=simple
 PIDFile=<%= install_dir %>/run/agent.pid
 User=dd-agent
 Restart=on-failure
```

Comment:
This is a useful approach as it avoids running `daemon-reload` and the service can be restarted right away. Good thinking! However, we should also include a test case to make sure that the file is read correctly and that the service restarts as expected.